engine/src/main/java/org/terasology/engine/core/subsystem/lwjgl/LwjglGraphics.java

```diff
@@ -70,6 +58,16 @@ public void registerCoreAssetTypes(ModuleAwareAssetTypeManager assetTypeManager)
         graphics.registerCoreAssetTypes(assetTypeManager);
     }
 
+    @Override
+    public void preInitialise(Context context) {
+        super.preInitialise(context);
+        GameEngine engine = context.get(GameEngine.class);
+
+        // Make sure NUI classes are registered as part of the engine module.
+        // FIXME: kludgy use of method outside of Interface
+        ((TerasologyEngine) engine).addToClassesOnClasspathsToAddToEngine(CanvasRenderer.class);
```

Comment:
FIXME: add this method to GameEngine interface, or figure out alternate interface.

Reply:
I'm calling this "add a NUI class to the engine module's classpath" thing from the LwjglGraphics subsystem.

My thinking was that it doesn't necessarily need to happen for headless mode, and LwjglGraphics is a thing you get when not in headless mode?

Lwjgl mostly doesn't depend on NUI, except it does already involve this CanvasRenderer, so it's not a new dependency.

Reply:
I also think it's weird for the subsystem to push this information to engine like this.

Engine already adds the subsystem classes themselves; maybe Subsystem should have some interface by which it can specify additional classes engine can pull in.